File: Algorithms/724. Find Pivot Index/php/solution.php
<?php
class Solution
{

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function pivotIndex($nums)
    {
        $sum = array_sum($nums);
        $leftSum = 0;
        foreach ($nums as $i => $num) {
            if ($leftSum == $sum - $leftSum - $num) {
                return $i;
            }
            $leftSum += $num;
        }
        return -1;
    }
}

File: Algorithms/724. Find Pivot Index/kotlin/solution.kt
class Solution {
    fun pivotIndex(nums: IntArray): Int {
        val sum = nums.sum()
        var leftSum = 0
        for ((i, num) in nums.withIndex()) {
            if (leftSum == (sum - leftSum - num)) {
                return i
            }
            leftSum += num
        }
        return -1
    }
}
